src/main/scala/com/typesafe/sbteclipse/core/EclipseAutoPlugin.scala

```diff
@@ -0,0 +1,25 @@
+package com.typesafe.sbteclipse.plugin
+
+import sbt.{Setting, AutoPlugin}
+
+import com.typesafe.sbteclipse.core.{EclipsePlugin=>EclipseCorePlugin}
+
+import scala.xml.Node
+
+/** Migrates eclipse to be an autoplugin. */
+object EclipsePlugin extends AutoPlugin {
+  override def requires = sbt.plugins.JvmPlugin
+  override def trigger = allRequirements
+  // Auto import semantics users expect today.
```

Comment:
I wouldn't worry about backwards compatibility too much. The only code that typically depends on this is the user's build and that's easy to change. It's when you've got another library depending on this one in between that things get stickier